src/h3lib/lib/algos.c

```diff
@@ -813,6 +817,58 @@ H3Error _getEdgeHexagons(const GeoLoop *geoloop, int64_t numHexagons, int res,
     return E_SUCCESS;
 }
 
+/**
+ * Returns true if the given cell is in the polygon using the containment
+ * mode flags.
+ */
+bool cellInPolygon(H3Index hex, const GeoPolygon *geoPolygon,
+                   const BBox *bboxes, uint32_t flags) {
+    if ((flags & FLAG_ANY_VERTEX) == FLAG_ANY_VERTEX) {
```

Comment:
Does this style of check make sense if we aren't using 1 bit per flag? If we are using 1 bit per flag, do we need the equality check? (Maybe this is a standard flag pattern I'm just not following?)

Reply:
Indeed this check is incorrect and needs to be fixed.